flytekit/interaction/click_types.py

```diff
@@ -307,9 +309,29 @@ def convert(
 
         parsed_value = self._parse(value, param)
 
+        def has_nested_dataclass(t: typing.Type) -> bool:
```

Comment:
Add some comment for this function. If I pass a dataclass (not flyte type) to this function, it will return `True` here as well. That seems like not correct.